src/braket/aws/aws_quantum_task.py

```diff
@@ -236,7 +236,8 @@ def result(self) -> Union[GateModelQuantumTaskResult, AnnealingQuantumTaskResult
         Consecutive calls to this method return a cached result from the preceding request.
         """
         try:
-            return asyncio.get_event_loop().run_until_complete(self.async_result())
+            async_result = self.async_result()
```

Comment:
For learning purposes, what do you mean by "proper setup" in the PR description, what was the error message?

Reply:
I got the error message
RuntimeError: There is no current event loop in thread 'ThreadPoolExecutor-0_0'
when calling task.result() from a worker thread.

If there is no existing event loop, then asyncio.get_event_loop() only creates a new one if the OS thread is main (documentation: https://docs.python.org/3/library/asyncio-eventloop.html?highlight=asyncio%20set_event_loop#asyncio.get_event_loop). In the code for async_result() we create a new event loop anyway, but we need to separate that into a separate line to make sure that gets called first.

Updated PR description for clarity.